File: Backend-Java/src/main/java/org/example/DAL/entities/ProductImageEntity.java
package org.example.DAL.entities;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Table(name="tbl_product_images")
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ProductImageEntity extends BaseEntity {
    @ManyToOne
    @JoinColumn(name="product_id", nullable = false)
    private ProductEntity product;
}

File: Backend-C#/Data.MyShop/Entities/ProductImageEntity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.MyShop.Entities
{
    [Table("tbl_ProductImages")]
    public class ProductImageEntity : BaseEntity<int>
    {
        public ProductEntity Product { get; set; }

        [ForeignKey(nameof(Product))]
        public int? ProductId { get; set; }

    }
}
